diff -up dwl-orig/config.h dwl-fork/config.h
--- dwl-orig/config.h	2023-02-14 15:15:15.476585514 +0100
+++ dwl-fork/config.h	2023-02-14 15:03:43.892067568 +0100
@@ -1,10 +1,10 @@
 /* appearance */
 static const int sloppyfocus               = 1;  /* focus follows mouse */
 static const int bypass_surface_visibility = 0;  /* 1 means idle inhibitors will disable idle tracking even if it's surface isn't visible  */
-static const unsigned int borderpx         = 1;  /* border pixel of windows */
-static const float rootcolor[]             = {0.3, 0.3, 0.3, 1.0};
-static const float bordercolor[]           = {0.5, 0.5, 0.5, 1.0};
-static const float focuscolor[]            = {1.0, 0.0, 0.0, 1.0};
+static const unsigned int borderpx         = 2;  /* border pixel of windows */
+static const float rootcolor[]             = {0.45, 0.88, 0.95, 1.0};
+static const float bordercolor[]           = {0.45, 0.88, 0.95, 1.0};
+static const float focuscolor[]            = {0.1, 0.6, 0.65, 1.0};
 /* To conform the xdg-protocol, set the alpha to zero to restore the old behavior */
 static const float fullscreen_bg[]         = {0.1, 0.1, 0.1, 1.0};
 
@@ -22,9 +22,9 @@ static const Rule rules[] = {
 /* layout(s) */
 static const Layout layouts[] = {
 	/* symbol     arrange function */
-	{ "[]=",      tile },
-	{ "><>",      NULL },    /* no layout function means floating behavior */
-	{ "[M]",      monocle },
+	{ "[D]",      dwindle }, /* requires fibonnaci patch */
+        { "[M]",      monocle },      
+        { "[F]",      NULL }, /* means floating*/
 };
 
 /* monitors */
@@ -104,55 +104,63 @@ static const enum libinput_config_tap_bu
 #define SHCMD(cmd) { .v = (const char*[]){ "/bin/sh", "-c", cmd, NULL } }
 
 /* commands */
-static const char *termcmd[] = { "foot", NULL };
-static const char *menucmd[] = { "bemenu-run", NULL };
+static const char *term[] = { "kitty", NULL };
+static const char *browser[] = { "firefox", NULL };
+
+static const char *menu[] = { "rofi", "-show", "drun", NULL };
+static const char *power[] = { "rofi", "-show", "power-menu", "-modi", "power-menu:/~/.config/rofi/rofi-power-menu", NULL };
+static const char *help[] = { "rofi", "-show", "Pomoc", "-modi", "Pomoc:/~/.config/rofi/rofi-help-tab", NULL };
+
+static const char *prnscr[] = { "/bin/sh", "-c", "grim -g \"$(slurp)\"", NULL};
+
+static const char *volp[] = { "pamixer", "-i", "5", NULL };
+static const char *volm[] = { "pamixer", "-d", "5", NULL };
+static const char *tmic[] = { "pamixer", "--source", "alsa_input.pci-0000_0a_00.3.analog-stereo", "-t", NULL };
+
 
 static const Key keys[] = {
-	/* Note that Shift changes certain key codes: c -> C, 2 -> at, etc. */
-	/* modifier                  key                 function        argument */
-	{ MODKEY,                    XKB_KEY_p,          spawn,          {.v = menucmd} },
-	{ MODKEY|WLR_MODIFIER_SHIFT, XKB_KEY_Return,     spawn,          {.v = termcmd} },
-	{ MODKEY,                    XKB_KEY_j,          focusstack,     {.i = +1} },
-	{ MODKEY,                    XKB_KEY_k,          focusstack,     {.i = -1} },
-	{ MODKEY,                    XKB_KEY_i,          incnmaster,     {.i = +1} },
-	{ MODKEY,                    XKB_KEY_d,          incnmaster,     {.i = -1} },
-	{ MODKEY,                    XKB_KEY_h,          setmfact,       {.f = -0.05} },
-	{ MODKEY,                    XKB_KEY_l,          setmfact,       {.f = +0.05} },
-	{ MODKEY,                    XKB_KEY_Return,     zoom,           {0} },
-	{ MODKEY,                    XKB_KEY_Tab,        view,           {0} },
-	{ MODKEY|WLR_MODIFIER_SHIFT, XKB_KEY_C,          killclient,     {0} },
-	{ MODKEY,                    XKB_KEY_t,          setlayout,      {.v = &layouts[0]} },
-	{ MODKEY,                    XKB_KEY_f,          setlayout,      {.v = &layouts[1]} },
-	{ MODKEY,                    XKB_KEY_m,          setlayout,      {.v = &layouts[2]} },
-	{ MODKEY,                    XKB_KEY_space,      setlayout,      {0} },
-	{ MODKEY|WLR_MODIFIER_SHIFT, XKB_KEY_space,      togglefloating, {0} },
-	{ MODKEY,                    XKB_KEY_e,         togglefullscreen, {0} },
-	{ MODKEY,                    XKB_KEY_0,          view,           {.ui = ~0} },
-	{ MODKEY|WLR_MODIFIER_SHIFT, XKB_KEY_parenright, tag,            {.ui = ~0} },
-	{ MODKEY,                    XKB_KEY_comma,      focusmon,       {.i = WLR_DIRECTION_LEFT} },
-	{ MODKEY,                    XKB_KEY_period,     focusmon,       {.i = WLR_DIRECTION_RIGHT} },
-	{ MODKEY|WLR_MODIFIER_SHIFT, XKB_KEY_less,       tagmon,         {.i = WLR_DIRECTION_LEFT} },
-	{ MODKEY|WLR_MODIFIER_SHIFT, XKB_KEY_greater,    tagmon,         {.i = WLR_DIRECTION_RIGHT} },
-	TAGKEYS(          XKB_KEY_1, XKB_KEY_exclam,                     0),
-	TAGKEYS(          XKB_KEY_2, XKB_KEY_at,                         1),
-	TAGKEYS(          XKB_KEY_3, XKB_KEY_numbersign,                 2),
-	TAGKEYS(          XKB_KEY_4, XKB_KEY_dollar,                     3),
-	TAGKEYS(          XKB_KEY_5, XKB_KEY_percent,                    4),
-	TAGKEYS(          XKB_KEY_6, XKB_KEY_asciicircum,                5),
-	TAGKEYS(          XKB_KEY_7, XKB_KEY_ampersand,                  6),
-	TAGKEYS(          XKB_KEY_8, XKB_KEY_asterisk,                   7),
-	TAGKEYS(          XKB_KEY_9, XKB_KEY_parenleft,                  8),
-	{ MODKEY|WLR_MODIFIER_SHIFT, XKB_KEY_Q,          quit,           {0} },
+        /* Note that Shift changes certain key codes: c -> C, 2 -> at, etc. */
+        /* modifier                  key                 function        argument */
+        { MODKEY,                    XKB_KEY_space,      spawn,          {.v = menu} },
+        { MODKEY,                    XKB_KEY_a,          spawn,          {.v = term} },
+        { MODKEY,                    XKB_KEY_f,          spawn,          {.v = browser} },
+        { MODKEY,                    XKB_KEY_s,          spawn,          {.v = power} },
+        { MODKEY,                    XKB_KEY_h,          spawn,          {.v = help} },
+        { MODKEY,                    XKB_KEY_p,          spawn,          {.v = prnscr} },
+
+        { MODKEY,                    XKB_KEY_k,          killclient,     {0} },
+
+        { MODKEY|WLR_MODIFIER_ALT,   XKB_KEY_t,          setlayout,      {.v = &layouts[0]} },
+        { MODKEY|WLR_MODIFIER_ALT,   XKB_KEY_f,          setlayout,      {.v = &layouts[1]} },
+        { MODKEY|WLR_MODIFIER_ALT,   XKB_KEY_m,          setlayout,      {.v = &layouts[2]} },
+
+        { MODKEY,                    XKB_KEY_v,          togglefloating, {0} },
+        { MODKEY,                    XKB_KEY_l,          togglefullscreen, {0} },
+
+        { MODKEY|WLR_MODIFIER_ALT,   XKB_KEY_q,          quit,           {0} },
+
+        TAGKEYS(          XKB_KEY_q, XKB_KEY_Q,                           0),
+        TAGKEYS(          XKB_KEY_w, XKB_KEY_W,                           1),
+        TAGKEYS(          XKB_KEY_e, XKB_KEY_E,                           2),
+        TAGKEYS(          XKB_KEY_r, XKB_KEY_R,                           3),
+        TAGKEYS(          XKB_KEY_t, XKB_KEY_T,                           4),
 
-	/* Ctrl-Alt-Backspace and Ctrl-Alt-Fx used to be handled by X server */
-	{ WLR_MODIFIER_CTRL|WLR_MODIFIER_ALT,XKB_KEY_Terminate_Server, quit, {0} },
+        /* Ctrl-Alt-Backspace and Ctrl-Alt-Fx used to be handled by X server */
+        { WLR_MODIFIER_CTRL|WLR_MODIFIER_ALT,XKB_KEY_Terminate_Server, quit, {0} },
 #define CHVT(n) { WLR_MODIFIER_CTRL|WLR_MODIFIER_ALT,XKB_KEY_XF86Switch_VT_##n, chvt, {.ui = (n)} }
-	CHVT(1), CHVT(2), CHVT(3), CHVT(4), CHVT(5), CHVT(6),
-	CHVT(7), CHVT(8), CHVT(9), CHVT(10), CHVT(11), CHVT(12),
+        CHVT(1), CHVT(2), CHVT(3), CHVT(4), CHVT(5), CHVT(6),
+        CHVT(7), CHVT(8), CHVT(9), CHVT(10), CHVT(11), CHVT(12),
 };
 
 static const Button buttons[] = {
-	{ MODKEY, BTN_LEFT,   moveresize,     {.ui = CurMove} },
-	{ MODKEY, BTN_MIDDLE, togglefloating, {0} },
-	{ MODKEY, BTN_RIGHT,  moveresize,     {.ui = CurResize} },
+        { MODKEY, BTN_LEFT,     moveresize,     {.ui = CurMove} },
+        { MODKEY, BTN_MIDDLE,   togglefloating, {0} },
+        { MODKEY, BTN_RIGHT,    moveresize,     {.ui = CurResize} },
+
+        /*{ MODKEY, BTN_SIDE,   moveresize,     {.ui = CurResize} },
+        { MODKEY, BTN_EXTRA,    moveresize,     {.ui = CurResize} },*/
+
+        { 0, BTN_MIDDLE,        spawn,          {.v = tmic} },
+        { 0, BTN_SIDE,          spawn,          {.v = volm} },
+        { 0, BTN_EXTRA,         spawn,          {.v = volp} },
 };
Společné podadresáře: dwl-orig/.git a dwl-fork/.git
Společné podadresáře: dwl-orig/.github a dwl-fork/.github
Společné podadresáře: dwl-orig/protocols a dwl-fork/protocols
